File: server-aspnet-fastendpoints/src/Org.OpenAPITools/Models/ProfileResponse.cs
namespace Org.OpenAPITools.Models;


/// <summary>
/// 
/// </summary>
public class ProfileResponse 
{
    public bool Valid { get; set; }
    public string Message { get; set; }
    public double VarVersion { get; set; }
    public bool SerializeNulls { get; set; }
    public long StartTimeLog { get; set; }
    public string ErrorCode { get; set; }
    public List<NameStringValueResponse> Request { get; set; }
    public long LoginAccountId { get; set; }
    public bool Validated { get; set; }
    public AppInfoResponse AppInfo { get; set; }
    public bool CanViewAppInfo { get; set; }
    public bool CanViewFriendInfo { get; set; }
    public bool CanViewProfileInfo { get; set; }
    public long FlagCount { get; set; }
    public ConnectionInfoResponse FriendInfo { get; set; }
    public bool IsBlocked { get; set; }
    public bool IsProfileOwner { get; set; }
    public bool MetFlagThreshold { get; set; }
    public ProfileInfoResponse ProfileInfo { get; set; }
    public bool Trusted { get; set; }
    public bool Following { get; set; }
    public bool FriendRequested { get; set; }
    public bool FriendRequestPending { get; set; }
    public bool Blocked { get; set; }
    public bool Flagged { get; set; }
    public bool ProfileOwner { get; set; }
    public bool Friend { get; set; }
    public string Returning { get; set; }
}

File: server-scala-play-server/app/model/ProfileResponse.scala
package model

import play.api.libs.json._

/**
  * Represents the Swagger definition for ProfileResponse.
  */
@javax.annotation.Generated(value = Array("org.openapitools.codegen.languages.ScalaPlayFrameworkServerCodegen"), date = "2026-02-09T20:57:39.376804970Z[Etc/UTC]", comments = "Generator version: 7.20.0-SNAPSHOT")
case class ProfileResponse(
  valid: Option[Boolean],
  message: Option[String],
  version: Option[Double],
  serializeNulls: Option[Boolean],
  startTimeLog: Option[Long],
  errorCode: Option[String],
  request: Option[List[NameStringValueResponse]],
  loginAccountId: Option[Long],
  validated: Option[Boolean],
  appInfo: Option[AppInfoResponse],
  canViewAppInfo: Option[Boolean],
  canViewFriendInfo: Option[Boolean],
  canViewProfileInfo: Option[Boolean],
  flagCount: Option[Long],
  friendInfo: Option[ConnectionInfoResponse],
  isBlocked: Option[Boolean],
  isProfileOwner: Option[Boolean],
  metFlagThreshold: Option[Boolean],
  profileInfo: Option[ProfileInfoResponse],
  trusted: Option[Boolean],
  following: Option[Boolean],
  friendRequested: Option[Boolean],
  friendRequestPending: Option[Boolean],
  blocked: Option[Boolean],
  flagged: Option[Boolean],
  profileOwner: Option[Boolean],
  friend: Option[Boolean],
  returning: Option[String]
)

object ProfileResponse {
  implicit lazy val profileResponseJsonFormat: Format[ProfileResponse] = Json.format[ProfileResponse]
}

